server/service/osquery_utils/gen_queries_doc.go

```diff
@@ -0,0 +1,71 @@
+//go:build ignore
+// +build ignore
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/fleetdm/fleet/v4/server/config"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/service/osquery_utils"
+)
+
+func main() {
+	detailQueriesMap := osquery_utils.GetDetailQueries(context.Background(), config.FleetConfig{
+		Vulnerabilities: config.VulnerabilitiesConfig{
+			DisableWinOSVulnerabilities: false,
+		},
+		App: config.AppConfig{
+			EnableScheduledQueryStats: true,
+		},
+	}, &fleet.Features{
+		EnableSoftwareInventory: true,
+		EnableHostUsers:         true,
+	})
+	var b strings.Builder
+
+	b.WriteString(`<!-- DO NOT EDIT. This document is automatically generated. -->
+# Detail Queries Summary
+
+Following is a summary of the detail queries hardcoded in Fleet used to populate the device details:
+
+`)
+
+	type queryInfo struct {
+		name        string
+		detailQuery osquery_utils.DetailQuery
+	}
+	var detailQueries []queryInfo
+	for name, detailQuery := range detailQueriesMap {
+		detailQueries = append(detailQueries, queryInfo{name: name, detailQuery: detailQuery})
+	}
+	sort.Slice(detailQueries, func(i, j int) bool {
+		return detailQueries[i].name < detailQueries[j].name
+	})
+
+	for _, q := range detailQueries {
+		fmt.Fprintf(&b, "## %s\n\n", q.name)
+		platforms := strings.Join(q.detailQuery.Platforms, ", ")
+		if len(q.detailQuery.Platforms) == 0 {
+			platforms = "all"
+		}
+		fmt.Fprintf(&b, "- Platforms: %s\n\n", platforms)
+		if q.detailQuery.Discovery != "" {
+			fmt.Fprintf(&b, "- Discovery query:\n```sql\n%s\n```\n\n", q.detailQuery.Discovery)
+		}
+		fmt.Fprintf(&b, "- Query:\n```sql\n%s\n```\n\n", q.detailQuery.Query)
```

Comment:
curious: have you checked how the page looks like? I'm guessing it will depend on the markdown parser we use and the library/styles we use to render the code fences, but I have noticed that indentation is of the SQL statements uneven. Is this noticeable once rendered? does it matter? (my guess would be that the indentation is kept)

if so, should we

Reply:
Am all ears. Maybe we can start with this and iterate if it doesn't look good? (It's currently a new markdown file not linked by other documents, much like https://fleetdm.com/docs/using-fleet/audit-activities#audit-activities).

Reply:
gotcha, I was worried about how queries like `mdm_windows` would look like (look at the indentation) but it actually doesn't look so bad.

Having said that, I have found a bug!

The table of contents on the right is incomplete and the rendered content looks wrong:

![image](https://user-images.githubusercontent.com/4419992/213541380-eded4c8a-a552-4240-9868-b660ac764f6a.png)

I was able to fix it with this diff (both the trimming and the extra newlines are required):

```diff
diff --git a/server/service/osquery_utils/gen_queries_doc.go b/server/service/osquery_utils/gen_queries_doc.go
index 199b13131..e6f039dc0 100644
--- a/server/service/osquery_utils/gen_queries_doc.go
+++ b/server/service/osquery_utils/gen_queries_doc.go
@@ -59,9 +59,9 @@ Following is a summary of the detail queries hardcoded in Fleet used to populate
                }
                fmt.Fprintf(&b, "- Platforms: %s\n\n", platforms)
                if q.detailQuery.Discovery != "" {
-                       fmt.Fprintf(&b, "- Discovery query:\n```sql\n%s\n```\n\n", q.detailQuery.Discovery)
+                       fmt.Fprintf(&b, "- Discovery query:\n\n```sql\n%s\n```\n\n", strings.TrimSpace(q.detailQuery.Discovery))
                }
-               fmt.Fprintf(&b, "- Query:\n```sql\n%s\n```\n\n", q.detailQuery.Query)
+               fmt.Fprintf(&b, "- Query:\n\n```sql\n%s\n```\n\n", strings.TrimSpace(q.detailQuery.Query))
        }

        b.WriteString(`
```

Reply:
Thanks for this! 🙌  How did you test this?

Reply:
np! after running `make generate-doc`:

```sh
cd website && sails run scripts/build-static-content.js --skipGithubRequests && sails lift
```

you will have to follow the readme in the website to set up your dependencies, but it might just be `yarn install` (I don't remember!)

Reply:
Works! Thanks!